thrust/thrust/iterator/zip_iterator.h

```diff
@@ -195,6 +195,15 @@ class _CCCL_DECLSPEC_EMPTY_BASES zip_iterator : public detail::make_zip_iterator
       : m_iterator_tuple(iterator_tuple)
   {}
 
+  //! This constructor creates a new \p zip_iterator from a pack of iterators.
+  //!
+  //! \param iterators A pack of iterators to zip.
+  template <typename... Iterators,
+            ::cuda::std::enable_if_t<(::cuda::std::input_or_output_iterator<Iterators> && ...), int> = 0>
```

Comment:
This should also be SFINAEd by checking that we can actually construct the IterTuple